src/compiler/tracing.ts

```diff
@@ -245,17 +284,26 @@ namespace ts { // eslint-disable-line one-namespace-per-file
                     intrinsicName: (type as any).intrinsicName,
                     symbolName: symbol?.escapedName && unescapeLeadingUnderscores(symbol.escapedName),
                     recursionId: recursionToken,
+                    isTuple: objectFlags & ObjectFlags.Tuple ? true : undefined,
                     unionTypes: (type.flags & TypeFlags.Union) ? (type as UnionType).types?.map(t => t.id) : undefined,
                     intersectionTypes: (type.flags & TypeFlags.Intersection) ? (type as IntersectionType).types.map(t => t.id) : undefined,
                     aliasTypeArguments: type.aliasTypeArguments?.map(t => t.id),
                     keyofType: (type.flags & TypeFlags.Index) ? (type as IndexType).type?.id : undefined,
                     ...indexedAccessProperties,
                     ...referenceProperties,
                     ...conditionalProperties,
-                    firstDeclaration: firstDeclaration && {
+                    ...substitutionProperties,
+                    ...reverseMappedProperties,
+                    ...evolvingArrayProperties,
+                    destructuringPattern: destructuringPatternFile && {

```

Comment:
Picky: looks a bit messy since there's a discrepancy between `reverseMappedXXX` and `destructuringPattern.XXX` (ie, the former could also be a nested object).

Reply:
I think it's clean enough with the new helper.